src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -157,7 +157,7 @@ protected virtual NullableWalker.SnapshotManager GetSnapshotManager()
         internal ImmutableDictionary<Symbol, Symbol> GetRemappedSymbols()
         {
             EnsureNullabilityAnalysisPerformedIfNecessary();
-            Debug.Assert(_lazyRemappedSymbols is object || !Compilation.NullableSemanticAnalysisEnabled);
+            Debug.Assert(_lazyRemappedSymbols is object || this is AttributeSemanticModel || !Compilation.NullableSemanticAnalysisEnabled);
```

Comment:
>Debug.Assert(_lazyRemappedSymbols is object || this is AttributeSemanticModel || !Compilation.NullableSemanticAnalysisEnabled); [](start = 12, length = 127)

It feels like the assert is too relaxed now. Looking at it, one might get a wrong idea that it is always Ok for an AttributeSemanticModel to not have a map. However, I believe an AttributeSemanticModel for an attribute within a local function (on parameter or type parameter) is expected to have a map. If that is correct, consider tightening the assert to reflect that.